src/track/track.cpp

```diff
@@ -924,15 +924,16 @@ Track::ImportStatus Track::importBeats(
     }
     DEBUG_ASSERT(!m_pBeatsImporterPending);
     m_pBeatsImporterPending = pBeatsImporter;
-    if (m_streamInfo) {
+    if (m_pBeatsImporterPending->isEmpty()) {
+        // Just return the current import status without clearing any
+        // existing cue points.
+        m_pBeatsImporterPending.reset();
```

Comment:
Except the reordering, I also had to add this line, otherwise the debug assertion in line 925 would complain when importing the metadata from file tags twice.